components/match2/wikis/dota2/match_summary.lua

```diff
@@ -299,12 +300,13 @@ function CustomMatchSummary._opponentHeroesDisplay(opponentHeroesData, numberOfH
 	local color = opponentHeroesData.side or ''
 
 	for index = 1, numberOfHeroes do
+		local hero = opponentHeroesData[index]
 		local heroDisplay = mw.html.create('div')
 			:addClass('brkts-popup-side-color-' .. color)
 			:addClass('brkts-popup-side-hero')
 			:addClass('brkts-popup-side-hero-hover')
-			:node(HeroIcon._getImage{
-				hero = opponentHeroesData[index],
+			:node(Logic.isEmpty(hero) and NO_HERO or CharacterIcon.Icon{
```

Comment:
**option 1** (my prefered one since it makes the default icon available in other places easily):
add a `default` character (transparent icon) to the data and use `CharacterIcon.Icon{character = hero, size = SIZE_HERO} or CharacterIcon{character = default, size = SIZE_HERO})` here

**option 2**:
`CharacterIcon.Icon{character = hero, size = SIZE_HERO} or NO_HERO`
removes the need for the empty check since the characterIcon module already does it

Reply:
Option 1 then, we will need to display the character in more than one place anyway

Reply:
What you think of having a constant like `local NO_HERO = CharacterIcon.Icon{character = 'default', size = SIZE_HERO}` and then use it like you sugested:

```suggestion
			:node(CharacterIcon.Icon{character = hero, size = SIZE_HERO} or NO_HERO)
```

Reply:
```suggestion
			:node(CharacterIcon.Icon{character = hero or NO_HERO, size = SIZE_HERO})
```
this should do the same if you set `local NO_HERO = 'default'`

Reply:
Oui